### DISCUSSION TITLE
how rendering plane or floor in sceneView??

### DISCUSSION THREAD

**Discussion by JooHyun-plask** [sceneview/sceneform-android#51] [Q&A]
hi i develop display 3d object(non-ar) using SceneView

for now i render 3d object(glb) and play animation 

but i need plane under my 3d object

i see examples but all samples are using arFragment for ar

how can i render plane please help me

i rendered like this

`
public void renderLocalObject(){
        ModelRenderable.builder()
                .setSource(this, Uri.parse(localModel))
                .setIsFilamentGltf(true)
                .build()
                .thenAccept(modelRenderable -> {
                    addNodeToScene(modelRenderable);
                })
                .exceptionally(throwable -> {
                    String message = throwable.getMessage();

                    Handler mainHandler = new Handler(Looper.getMainLooper());
                    Runnable myRunnable = (Runnable)(new Runnable(){

                        @Override
                        public void run() {
                            dialog.getBuilder().content(message)
                                    .positiveText("")
                                    .canceledOnTouchOutside(false)
                                    .autoDismiss(false)
                                    .onPositive(new MaterialDialog.SingleButtonCallback() {
                                        @Override
                                        public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
                                            finish();
                                        }
                                    }).show();
                        }
                    });
                    mainHandler.post(myRunnable);
                    return null;
                });
    }
`

`
public void addNodeToScene(ModelRenderable model){
        if (mSceneView != null ){
            double y = 0.0;
            TransformationSystem transformationSystem = makeTransformationSystem();
            DragTransformableNode dragTransformableNode = new DragTransformableNode(3.0F, transformationSystem);

           
            dragTransformableNode.setLookDirection(new Vector3(1.0F, 0.0F, 2.0F));
            dragTransformableNode.setRenderable((Renderable) model);

            
            dragTransformableNode.setLocalScale(new Vector3(2.0F, 2.0F, 2.0F));
            dragTransformableNode.setLocalPosition(new Vector3(0.0F, -1.5F, 0.0F));
            Log.e(TAG, "model default scale : "+dragTransformableNode.getLocalScale().toString());
            Log.e(TAG, "model default position : "+dragTransformableNode.getLocalPosition().toString());

            Objects.requireNonNull(mSceneView.getScene().getSunlight()).setEnabled(true);
            mSceneView.getScene().addChild((Node) dragTransformableNode);

            if (Objects.requireNonNull(dragTransformableNode.getRenderableInstance()).hasAnimations()){
                dragTransformableNode.getRenderableInstance().animate(true).start();
            }
            dragTransformableNode.select();
            mSceneView.getScene().addOnPeekTouchListener(new Scene.OnPeekTouchListener() {
                @Override
                public void onPeekTouch(HitTestResult hitTestResult, MotionEvent motionEvent) {
                    transformationSystem.onTouch(hitTestResult, motionEvent);
                }
            });
        }
    }
`

and i hope like this

![62502918-ba538280-b7a6-11e9-91e2-a87f37867775](https://user-images.githubusercontent.com/71476797/116676877-7ecd0300-a9e2-11eb-8e95-442f0cf12473.jpeg)

**Comment by ThomasGorisse** [comment]
What is your question ?

**Reply by JooHyun-plask** [reply]
sorry my explain strange....

is it possible make floor?? like that image

**Comment by ThomasGorisse** [comment]
Have a look at `ShapeFactory` class
